static/src/javascripts/projects/commercial/modules/third-party-tags.js

```diff
@@ -52,28 +52,29 @@ const addScripts = (tags) => {
 	});
 
 	if (hasScriptsToInsert) {
-		fastdom.mutate(() => {
+		return fastdom.mutate(() => {
 			if (ref && ref.parentNode) {
 				ref.parentNode.insertBefore(frag, ref);
 			}
 		});
 	}
+	return Promise.resolve();
 };
 
-const insertScripts = (
+const insertScripts = async (
 	advertisingServices,
 	performanceServices, // performanceServices always run
 ) => {
-	addScripts(performanceServices);
-	void getInitialConsentState().then((state) => {
-		const consentedAdvertisingServices = advertisingServices.filter(
-			(script) => getConsentFor(script.name, state),
-		);
+	await addScripts(performanceServices);
+	const state = await getInitialConsentState();
+	const consentedAdvertisingServices = advertisingServices.filter((script) =>
+		getConsentFor(script.name, state),
+	);
 
-		if (consentedAdvertisingServices.length > 0) {
-			addScripts(consentedAdvertisingServices);
-		}
-	});
+	if (consentedAdvertisingServices.length > 0) {
+		await addScripts(consentedAdvertisingServices);
+	}
+	return Promise.resolve(true);
```

Comment:
Do we want to await the third party adding of scripts here - would it be added to our metrics?

Reply:
Good catch, I hadn't realized that [line 57](https://github.com/guardian/frontend/blob/c6b898e7ecb3b1498e3d46bf4ce7bac8e734996b/static/src/javascripts/projects/commercial/modules/third-party-tags.js#L57) was adding the script tags without awaiting their execution. 
I think we do want to find a way to change `addScripts` so that the scripts are loaded asynchronously before resolving here.

Reply:
@arelra are you suggesting that we should not account for third-party loading times?

Maybe we can skip the `await`ing now, convert this to TS and look into using `loadScript` from `@guardian/libs`?

Reply:
Yep I'm just wondering if awaiting third party scripts makes sense wrt the loading sequence and our metrics?

I feel like they shouldn't be awaited but I'm not super clear on what our metrics intend to capture wrt first and third party - perhaps we need better clarity there?

Reply:
I like the idea of using `loadScript` as it returns a promise that we can await.
It feels like that change is in the scope of this PR though.

Reply:
I’d _much_ prefer converting to TypeScript first. But that’s me 🦺